server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -93,13 +103,22 @@ public MailboxFactory(MailboxManager mailboxManager) {
         return name;
     }
 
-    @VisibleForTesting Optional<MailboxId> getParentIdFromMailboxPath(MailboxPath mailboxPath, MailboxSession mailboxSession) throws MailboxException {
+    @VisibleForTesting Optional<MailboxId> getParentIdFromMailboxPath(MailboxPath mailboxPath, Optional<List<MailboxMetaData>> userMailboxesMetadata,
```

Comment:
No.

It needs refactoring to do so.

You have a difference between : 
 - I don't took time to retrieve mailboxes
 - There is no mailboxes